File: DataTierGeneratorPlusTester/Vb_BuiltIn_WcfServiceApp/Test_masterServiceExtensions.vb
Imports System.Runtime.CompilerServices
Imports System.Runtime.Serialization
Imports System.ServiceModel

'Namespace Vb_BuiltIn_WcfServiceApp 

	Friend Module Test_masterServiceExtensions 

		''' <summary>
		''' Converts from List of Test_masterInfo to List of Test_masterContract.
		''' </summary>
		<Extension()> _
		Friend Function ToBindingListViewOfContract(ByVal infoList As BindingListView(Of Test_masterInfo)) As BindingListView(Of Test_masterContract)  
			Dim returnValue As New BindingListView(Of Test_masterContract) 
			For Each info As Test_masterInfo In infoList
				returnValue.Add _
					( _
					info.ToTest_masterContract() _
					)
			Next
			Return returnValue
		End Function

		''' <summary>
		'''  Converts from Test_masterInfo to Test_masterContract.
		''' </summary>
		<Extension()> _
		Friend Function ToTest_masterContract(ByVal info As Test_masterInfo) As Test_masterContract
			Return New Test_masterContract( _
				info.id, _
				info.description, _
				info.notes, _
				info.someint, _
				info.someint_nullable, _
				info.somedate, _
				info.somedate_nullable, _
				info.somefloat, _
				info.somefloat_nullable, _
				info.somebool, _
				info.somebool_nullable _
			)
		End Function

		''' <summary>
		'''  Converts from Test_masterContract to Test_masterInfo.
		''' </summary>
		<Extension()> _
		Friend Function ToTest_masterInfo(ByVal info As Test_masterContract) As Test_masterInfo
			Return New Test_masterInfo( _
				info.id, _
				info.description, _
				info.notes, _
				info.someint, _
				info.someint_nullable, _
				info.somedate, _
				info.somedate_nullable, _
				info.somefloat, _
				info.somefloat_nullable, _
				info.somebool, _
				info.somebool_nullable _
			)
		End Function
	End Module
'End Namespace

File: DataTierGeneratorPlusTester/Cs_BuiltIn_WcfServiceApp/Test_masterServiceExtensions.cs
using System.Runtime.Serialization;
using System.ServiceModel;

namespace Cs_BuiltIn_WcfServiceApp 
{

	internal static class Test_masterServiceExtensions
	{

		/// <summary>
		/// Converts from List of Test_masterInfo to List of Test_masterContract.
		/// </summary>
		internal static BindingListView<Test_masterContract> ToBindingListViewOfContract(this BindingListView<Test_masterInfo> infoList) 
		{ 
			BindingListView<Test_masterContract> returnValue = new BindingListView<Test_masterContract>();
			foreach (Test_masterInfo info in infoList) 
			{
				returnValue.Add
					(
					info.ToTest_masterContract()
					);
			}
			return returnValue;
		}

		/// <summary>
		/// Converts from Test_masterInfo to Test_masterContract.
		/// </summary>
		internal static Test_masterContract ToTest_masterContract(this Test_masterInfo info) 
		{
			return new Test_masterContract
			(
				info.id,
				info.description,
				info.notes,
				info.someint,
				info.someint_nullable,
				info.somedate,
				info.somedate_nullable,
				info.somefloat,
				info.somefloat_nullable,
				info.somebool,
				info.somebool_nullable
			);
		}

		/// <summary>
		/// Converts from Test_masterContract to Test_masterInfo.
		/// </summary>
		internal static Test_masterInfo ToTest_masterInfo(this Test_masterContract info) 
		{
			return new Test_masterInfo
			(
				info.id,
				info.description,
				info.notes,
				info.someint,
				info.someint_nullable,
				info.somedate,
				info.somedate_nullable,
				info.somefloat,
				info.somefloat_nullable,
				info.somebool,
				info.somebool_nullable
			);
		}
	}
}
